gammapy/spectrum/spectrum_fit.py

```diff
@@ -293,12 +290,13 @@ def _run_sherpa_fit(self):
         thres_lo = self.energy_threshold_low.to('keV').value
         thres_hi = self.energy_threshold_high.to('keV').value
 
+        namedataset = []
         for i in range(len(ds.datasets)):
             datastack.notice_id(i + 1, thres_lo[i], thres_hi[i])
-
+            namedataset.append(i + 1)
         datastack.set_stat(self.statistic)
-        ds.fit()
-        datastack.covar()
+        ds.fit(*namedataset)
```

Comment:
I get the idea. At some point we should move to the real sherpa class interface anyways, so problems like this should go away ..